sorting/bead_sort.cpp

```diff
@@ -14,7 +14,7 @@ void beadSort(int *a, int len) {
 
     // allocating memory
     unsigned char *beads = new unsigned char[max * len];
-    memset(beads, 0, max * len);
+    memset(beads, 0, (__int64_t) max * len);
```

Comment:
here as well, the correct typecast would be to a `size_t`.
```suggestion
    memset(beads, 0, static_cast<size_t>(max) * len);
```